roles/cobbler_profile/tasks/import_distro.yml

```diff
@@ -0,0 +1,88 @@
+---
+# This profile will do all the work necessary to create a new distro/profile
+# pair in Cobbler.
+
+# Since this profile will be used several times in the same playbook,
+# mention the distro name each time.
+- name: Distro name
+  debug: var=distro_name
+
+- name: Load extra_distros from secrets
+  set_fact:
+      distros: "{{ distros.update(extra_distros|default({})) }}{{ distros}}"
+
+- name: Find distro settings
+  set_fact:
+      distro: "{{ distros[distro_name] }}"
+
+- name: Determine if distro profile exists
+  command: cobbler profile find --name {{ distro_name }}
+  # Skip if the iso field is empty; this allows us to mention distros with 
+  # ISOs that are internal, but leave the URL out.
+  when: distro.iso != ''
+  register: profile
+  ignore_errors: true
+  changed_when: false
+
+- name: Create ISO directory
+  file:
+      path: "{{ iso_dir }}"
+      state: directory
+  when: distro.iso != ''
+
+- name: Create ISO mountpoint
+  file:
+      path: "{{ iso_mount }}"
+      state: directory
+  when: distro.iso != ''
+
+- name: Set ISO name
+  set_fact:
+      iso_name: "{{ distro.iso.split('/')[-1] }}"
+  when: distro.iso != ''
+
+- name: Set ISO path
+  set_fact:
+      iso_path: "{{ iso_dir }}/{{ iso_name }}"
+  when: distro.iso != ''
+
+- name: Download ISO
+  get_url:
+      url={{ distro.iso }}
+      dest={{ iso_path }}
+  when: profile is defined and profile.stdout == ''
+  register: download
+
+- name: Mount ISO
+  command: mount -o loop {{ iso_path }} {{ iso_mount }}
+  when: download|changed
+  register: mount
+
+- name: Set arch
+  set_fact:
+      arch: "{{ distro.arch|default('x86_64') }}"
+  when: mount is defined and mount.rc == 0
+
+- name: Import the distro (also creates the profile)
```

Comment:
before import you might want to run 'cobbler signature update'.  I had to for it to pick up the new vivid information.  If I didn't the import would fail.